lib/charms/opensearch/v0/opensearch_users.py

```diff
@@ -313,33 +332,3 @@ def put_internal_user(self, user: str, hashed_pwd: str):
                 COSUser,
                 [{"op": "replace", "path": "/opendistro_security_roles", "value": roles}],
             )
-
-    def _remove_lingering_users(self, relation_users: Set[str]):
-        app_users = relation_users | OpenSearchUsers
-        try:
-            database_users = set(self.get_users().keys())
-        except OpenSearchUserMgmtError:
-            logger.error("failed to get users")
-            return
-
-        for username in database_users - app_users:
-            try:
-                self.remove_user(username)
-            except OpenSearchUserMgmtError:
-                logger.error(f"failed to remove user {username}")
-
-    def _remove_lingering_roles(self, roles: Set[str]):
```

Comment:
let's rollback

Reply:
Reworked, structure changed. We should discuss latest changes.